src/Network/Network/PrivateLinkService/PrivateLinkServiceProvider/GenericProvider.cs

```diff
@@ -70,36 +64,53 @@ private IAzureRestClient ServiceClient
         public PSPrivateEndpointConnection GetPrivateEndpointConnection(string resourceGroupName, string serviceName, string name)
         {
             string url = BuildPrivateEndpointConnectionURL(resourceGroupName, serviceName, name);
-            PrivateEndpointConnection connnection = ServiceClient.Operations.GetResouce<PrivateEndpointConnection>(url, _apiVersion);
+            PrivateEndpointConnection connnection = ServiceClient.Operations.GetResouce<PrivateEndpointConnection>(url, _configuration.ApiVersion);
             return ToPsPrivateEndpointConnection(connnection);
         }
 
         public List<PSPrivateEndpointConnection> ListPrivateEndpointConnections(string resourceGroupName, string serviceName)
         {
             var psPECs = new List<PSPrivateEndpointConnection>();
-            string url = BuildPrivateEndpointConnectionsURL(resourceGroupName, serviceName);
-            IPage<PrivateEndpointConnection> list = ServiceClient.Operations.GetResoucePage<Page<PrivateEndpointConnection>, PrivateEndpointConnection>(url, _apiVersion);
-            foreach (var pec in list)
-            {
-                var psPec = ToPsPrivateEndpointConnection(pec);
-                psPECs.Add(psPec);
-            }
-            while (list.NextPageLink != null)
+            if(_configuration.HasConnectionsURI)
             {
-                list = ServiceClient.Operations.GetResoucePage<Page<PrivateEndpointConnection>, PrivateEndpointConnection>(list.NextPageLink, null);
+                string url = BuildPrivateEndpointConnectionsURL(resourceGroupName, serviceName);
+                IPage<PrivateEndpointConnection> list = ServiceClient.Operations.GetResoucePage<Page<PrivateEndpointConnection>, PrivateEndpointConnection>(url, _configuration.ApiVersion);
                 foreach (var pec in list)
                 {
                     var psPec = ToPsPrivateEndpointConnection(pec);
                     psPECs.Add(psPec);
                 }
+                while (list.NextPageLink != null)
+                {
+                    list = ServiceClient.Operations.GetResoucePage<Page<PrivateEndpointConnection>, PrivateEndpointConnection>(list.NextPageLink, null);
+                    foreach (var pec in list)
+                    {
+                        var psPec = ToPsPrivateEndpointConnection(pec);
+                        psPECs.Add(psPec);
+                    }
+                }
+            } 
+            else
+            {
+                string url = BuildPrivateEndpointConnectionsOwnerURL(resourceGroupName, serviceName);
+                TrackedResource resource = ServiceClient.Operations.GetResouce<TrackedResource>(url, _configuration.ApiVersion);
+                if(resource.PrivateEndpointConnections != null)
```

Comment:
Suggest add a `?.` to avoid potential null ref error (for example if user inputs a wrong resource id, so that resource is not found)

```suggestion
                if(resource?.PrivateEndpointConnections != null)
```